src/Lynx/List.purs

```diff
@@ -0,0 +1,216 @@
+
+component :: forall f. Halogen.Component Halogen.HTML.HTML Query Input Output f
+component =
+  Halogen.component
+    { eval
+    , initialState
+    , receiver
+    , render
+    }
+
+eval :: forall f. Query ~> f
+eval query = eval query
+
+fromArgonaut ::
+  forall a.
+  Data.Argonaut.DecodeJson a =>
+  Data.Argonaut.EncodeJson a =>
+  Data.Codec.Argonaut.JsonCodec a
+fromArgonaut = Data.Codec.Argonaut.prismaticCodec from to Data.Codec.Argonaut.json
+  where
+  from :: Data.Argonaut.Json -> Maybe a
+  from json = case Data.Argonaut.decodeJson json of
+    Left _ -> Nothing
+    Right x -> Just x
+
+  to :: a -> Data.Argonaut.Json
+  to = Data.Argonaut.encodeJson
+
+-- | We provide a way to decode `argonaut`'s idea of JSON to a `View`.
+fromJSON :: Data.Argonaut.Json -> Either String View
+fromJSON json = case Data.Codec.Argonaut.decode view json of
+  Left err -> Left (Data.Codec.Argonaut.printJsonDecodeError err)
+  Right x -> Right x
+
+initialState :: Input -> State
+initialState input = do
+  columns <- Data.Traversable.traverse column input.columns
+  pure
+    { columns
+    }
+  where
+  column :: Column -> Either Lynx.Expr.EvalError RenderColumn
+  column column' = do
+    name <- Lynx.Expr.evalExpr lookup column'.name
+    pure
+      { name
+      }
+
+  lookup :: Lynx.Expr.Key -> Maybe Lynx.Expr.ExprType
+  lookup key = Nothing
+
+-- | This is the main thing that decides how to de/serialize our values.
+-- |
+-- | Something we run into from time to time with separate de/encoders is them
+-- | becoming out of sync.
+-- | It's a _mostly_ caught by the test suite,
+-- | but we can also allow the de/encoders to be correct by construction.
+jsonView :: Data.Codec.Argonaut.JsonCodec JSONView
+jsonView =
+  Data.Codec.Argonaut.object "Lynx.List.JSONView"
+    $ Data.Codec.Argonaut.recordProp (SProxy :: _ "columns") jsonColumns
+    $ Data.Codec.Argonaut.record
+  where
+  jsonColumn :: Data.Codec.Argonaut.JsonCodec JSONColumn
+  jsonColumn =
+    Data.Codec.Argonaut.object "Lynx.List.JSONColumn"
+      $ Data.Codec.Argonaut.recordProp (SProxy :: _ "name") fromArgonaut
+      $ Data.Codec.Argonaut.recordProp (SProxy :: _ "value") Data.Codec.Argonaut.string
+      $ Data.Codec.Argonaut.record
+
+  jsonColumns :: Data.Codec.Argonaut.JsonCodec (Array JSONColumn)
+  jsonColumns = Data.Codec.Argonaut.array jsonColumn
+
+receiver :: Input -> Maybe (Query Unit)
+receiver input = Nothing
+
+render :: forall a f. State -> Halogen.HTML a f
+render state' = case state' of
+  Left error ->
+    Ocelot.Block.Card.card_
+      [ Ocelot.Block.Format.p
+          [ Ocelot.HTML.Properties.css "text-red"
+          ]
+          [ evalError error
+          ]
+      ]
+  Right state ->
+    Ocelot.Block.Table.table_
+      [ Ocelot.Block.Table.row_ (map column state.columns)
+      ]
+  where
+  column :: RenderColumn -> Halogen.HTML a f
+  column column' =
+    Ocelot.Block.Table.header_
+      [ Halogen.HTML.text (Lynx.Expr.print column'.name)
+      ]
```

Comment:
I would think PS-3032 also involves rendering rows for a collection of `Array (Page a)` here, no?

Reply:
Where does this live? Is it part of the list `View` or is it part of the `RenderView`?

Reply:
So if I understand your setup, `View` and `RenderView` are analogous to `View Expr` and `View ExprType`, if we were to do things more similar to `Lynx.purs`, yeah? In that case we just need to add `value :: ExprType` (or `String`) to `RenderView`, and add an eva that goes from `View -> RenderView`.

Reply:
`View` and `RenderView` would be more like `Page Expr` and
```PureScript
type State
  = { activeTab :: String
    , evaled :: Either EvalError (Page ExprType)
    , expr :: Page Expr
    , fragment :: Fragment
    , idGenerator :: Aff String
    , values :: Map Key ExprType
    }
```
respectively. In any case, sounds like it belongs in the `RenderView`.